### PR TITLE
Adding support for message formatting in exception c'tors

### PR DESCRIPTION
**Pull request by whoisj**
Just simplifies the code. There's not a great reason to continually use `string.Format(...)` constantly.

Was limited to using the `public LibGit2SharpException(CultureInfo cultureInfo, string format, params object[] args)` to avoid a conflict with the `public LibGit2SharpException(string message, Exception innerException)` if I used `public LibGit2SharpException(string format, params object[] args)`.

### CODE REVIEW

**Comment by whoisj** [comment]
Looks like I packed up that commit incorrectly while tying to undo the auto-formatting mess VS keeps getting me into. Let me see what I did there :confounded:

**Comment by whoisj** [comment]
Nope - it actually looks like the NRE issue come back to haunt me. :cry:

**Comment by nulltoken** [comment]
> Nope - it actually looks like the NRE issue come back to haunt me.

Restarted the build

File: LibGit2Sharp/Diff.cs [hunk 1]
```diff
@@ -210,11 +210,13 @@ public virtual ContentChanges Compare(Blob oldBlob, Blob newBlob, CompareOptions
         {
             Func<DiffSafeHandle, object> builder;
 
-            if (!ChangesBuilders.TryGetValue(typeof (T), out builder))
```
Comment (nulltoken) [inline LibGit2Sharp/Diff.cs]: Any chance you could move the beautifications in a separate commit?

File: LibGit2Sharp/Diff.cs [hunk 2]
```diff
@@ -321,11 +323,13 @@ public virtual ContentChanges Compare(Blob oldBlob, Blob newBlob, CompareOptions
...
```
Comment (nulltoken) [inline LibGit2Sharp/Diff.cs]: Here as well

File: LibGit2Sharp/Diff.cs [hunk 3]
```diff
@@ -451,11 +455,13 @@ public virtual ContentChanges Compare(Blob oldBlob, Blob newBlob, CompareOptions
...
```
Comment (nulltoken) [inline LibGit2Sharp/Diff.cs]: Ditto

File: LibGit2Sharp/Repository.cs [hunk 4]
```diff
@@ -1546,14 +1544,14 @@ private MergeResult NormalMerge(GitAnnotatedCommitHandle[] annotatedCommits, Sig
             MergeResult mergeResult;
 
             var mergeOptions = new GitMergeOpts
-                {
-                    Version = 1,
-                    MergeFileFavorFlags = options.MergeFileFavor,
-                    MergeTreeFlags = options.FindRenames ? GitMergeTreeFlags.GIT_MERGE_TREE_FIND_RENAMES :
+            {
```
Comment (nulltoken) [inline LibGit2Sharp/Repository.cs]: Same here

**Comment by nulltoken** [comment]
I really like this proposal!

File: LibGit2Sharp/Diff.cs (hunk 1 above)
Comment (whoisj) [inline LibGit2Sharp/Diff.cs]: @#$%! I thought I got that all cleaned up :panda_face:

**Comment by nulltoken** [comment]
:sparkles: :sparkles: :sparkles: :sparkles:
